drainer/syncer.go

```diff
@@ -466,12 +466,16 @@ ForLoop:
 				schema, table string
 			)
 			sql := b.job.Query
-			schema, table, err = s.schema.getSchemaTableAndDelete(b.job.BinlogInfo.SchemaVersion)
-			if err != nil {
-				err = errors.Trace(err)
-				break ForLoop
-			}
-			if s.cfg.SyncDDL {
+
+			if b.job.BinlogInfo.SchemaVersion == 0 {
+				log.Info("skip ddl due to the failed ddl", zap.String("sql", sql), zap.Int64("commit ts", commitTS))
+				shouldSkip = true
+			} else if s.cfg.SyncDDL {
+				schema, table, err = s.schema.getSchemaTableAndDelete(b.job.BinlogInfo.SchemaVersion)
```

Comment:
I think we can't skip this step when `s.cfg.SyncDDL == false`. The `version2SchemaTable` map will keep increasing and cause an oom issue. What's more, `schema, table` info will be needed when we do the following steps.